### PR TITLE
Changed heap dumps to be optional only if path is specified

### PR DESCRIPTION
**Pull request by j4b0l**

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -42,8 +40,13 @@ export JAVA_OPTS="${JAVA_OPTS} ${JAVA_SECURITY_EGD}"
 export JAVA_OPTS="${JAVA_OPTS} ${POM_JAVA_OPTS}"
 export JAVA_OPTS="${JAVA_OPTS} ${CUSTOM_JAVA_OPTS}"
 
-if [[ ! -z ${DEBUG_PORT} ]]; then
-   export JAVA_OPTS="${JAVA_OPTS} -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=${DEBUG_PORT}" 
+if [[ ! -z "${DEBUG_PORT}" ]]; then
```
Comment (dawidmalina) [inline start.sh]: The same change should be proposed in https://github.com/oberthur/docker-generic-app/blob/master/simple-start.sh#L16

File: start.sh [hunk 2]
```diff
@@ -42,8 +40,13 @@ export JAVA_OPTS="${JAVA_OPTS} ${JAVA_SECURITY_EGD}"
...
+    export JAVA_OPTS="${JAVA_OPTS} -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=${DEBUG_PORT}"
+fi
+
+if [[ ! -z "${HEAP_DUMP_DIR}" ]]; then
+    export JAVA_OPTS="${JAVA_OPTS} -XX:+HeapDumpOnOutOfMemoryError"
+    export JAVA_OPTS="${JAVA_OPTS} -XX:HeapDumpPath=${HEAP_DUMP_DIR}"
```
Comment (dawidmalina) [inline start.sh]: I would say that this shall be configured to /dev/null to make sure that java itself will not dump memory to file.

File: start.sh (hunk 1 above)
Comment (j4b0l) [inline start.sh]: start_simple.sh looks like... well... simple file, without fancy functions like debug port or heap dump given by simple variable-exists switch. Is it a good candidate for start_simple.sh ?

**Comment by j4b0l** [comment]
@dawidmalina can you please check now? Also see my comment regarding simple script.

File: start.sh (hunk 2 above)
Comment (dawidmalina) [inline start.sh]: Ok so why do we need this
```
export JAVA_OPTS="${JAVA_OPTS} -XX:+ExitOnOutOfMemoryError"
```

File: start.sh (hunk 1 above)
Comment (dawidmalina) [inline start.sh]: Ok so why do we need this as this is ... simple
```
export JAVA_OPTS="${JAVA_OPTS} -XX:+ExitOnOutOfMemoryError"
export JAVA_OPTS="${JAVA_OPTS} -XX:+PrintGCDetails"
export JAVA_OPTS="${JAVA_OPTS} -XX:+PrintGCTimeStamps"
export JAVA_OPTS="${JAVA_OPTS} -XX:+PrintGCDateStamps"
```

Comment (j4b0l) [inline start.sh]: I think that will be better answered by author: @tomasare can you elaborate?

**Pull request merged**